src/duk_js_executor.c

```diff
@@ -2250,6 +2252,17 @@ DUK_INTERNAL void duk_js_execute_bytecode(duk_hthread *exec_thr) {
 		 * whenever a thread switch occurs by the DUK_HEAP_SWITCH_THREAD() macro.
 		 */
 #ifdef DUK_USE_INTERRUPT_COUNTER
+
+		if (thr->heap->dbg_processing) {
+			/* Don't interrupt if we're currently doing debug processing as
+			 * this may cause the debugger to be called recursively. Check required
+			 * for correct operation of throw intercept and other "exotic" halting
+			 * scenarios.
+			 */
+
+			goto skip_interrupt;
```

Comment:
Skipping the interrupt handler entirely will cause instruction counts based on cumulative interrupt counter values to be off sync and should be debug logged e.g. here: https://github.com/svaarala/duktape/blob/master/src/duk_js_executor.c#L1700-L1706

The guarantee that interrupt counter matches exactly to opcodes executed is a goal but it's not yet needed for anything specific (like sampling based profiling). So, this can be addressed in a cleanup branch too.

The fix would probably be to move the check where the recursive interrupt check is: https://github.com/svaarala/duktape/blob/master/src/duk_js_executor.c#L1716

Not sure if that causes other issues.